### ISSUE TITLE
Adding awesome-linters to the main `awesome` repository

### ISSUE THREAD

**Issue by mlcdf** [caramelomartins/awesome-linters#28]
(following PR #27)

I went looking at some PR on [sindresorhus/awesome](https://github.com/sindresorhus/awesome) and I stumbled upon this:

-----
# By submitting this pull request I confirm I've read and complied with the below requirements.
Failure to properly do so will just result in the pull request being closed and everyone's time wasted.  Please read it twice. Most people miss many things.

- I have read and understood the [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/contributing.md) and the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title. For example, `Add Name of List`, not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should include a short description about the list project/theme (not the list itself), and should be added at the bottom of the appropriate category.
- The list I added
	- **has been around for at least 30 days,**
	- is the result of hard work and is the best I could possibly produce,
	- is a non-generated Markdown file in a GitHub repo,
	- is not a duplicate,
	- includes a succinct description of the project/theme at the top ([example](https://github.com/willempienaar/awesome-quantified-self)),
	- includes a project logo/illustration (whenever possible) at the top right of the readme ([example](https://github.com/sindresorhus/awesome-electron)). (The logo should link to the project website or any relevant website. The logo should be high-DPI, so set it to maximum half the width of the original image.),
	- only has awesome items **with descriptions** *(Awesome lists are curations of the best, not of everything)*,
	- contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
	- has a Table of Contents section **named `Contents`** as the first section,
	- has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) *(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)* *(If you use a license badge, it should be SVG, not PNG)*,
	- has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
	- doesn't include a Travis badge,
	- has been checked for inconsistent formatting, spelling, and grammar:
		- Each link description starts with an uppercase character and ends with a period.
		- Drop all the `A` / `An` prefixes in the descriptions.
		- Consistent naming. For example, `Node.js`, not `NodeJS` or `node.js`.

----

In order to be listed in the `awesome` repo, we would need to add a little description for each items. Does this sounds good to you?

**Comment by caramelomartins** [comment]
Hello,

I'm sorry for the delay in the response. I wanted to think about it for a bit first before answering.

I started this list a long time ago just for me and some friends to have so point of reference for cool linters to use but after a while any problem must evolve and as I see more people contributing and staring it I understand that it has some purpose for other people also.

With that said, I think it'd be cool if we could get into that list but those rules might be hard to follow, let's tackle them:
- [x] **has been around for at least 30 days,**
- [x] is the result of hard work and is the best I could possibly produce,
- [x] is a non-generated Markdown file in a GitHub repo,
- [x] is not a duplicate,
- [ ] includes a succinct description of the project/theme at the top ([example](https://github.com/willempienaar/awesome-quantified-self)),
- [ ] ~includes a project logo/illustration (whenever possible) at the top right of the readme ([example](https://github.com/sindresorhus/awesome-electron)). (The logo should link to the project website or any relevant website. The logo should be high-DPI, so set it to maximum half the width of the original image.),~
- [ ] only has awesome items **with descriptions** *(Awesome lists are curations of the best, not of everything)*,
- [x] contains the [awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge) on the right side of the list heading,
- [ ] has a Table of Contents section **named `Contents`** as the first section,
- [x] has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license) *(Meaning something like CC0, NOT a code licence like MIT, BSD, Apache, etc)* *(If you use a license badge, it should be SVG, not PNG)*,
- [x] has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines),
- [x] doesn't include a Travis badge,
- [ ] has been checked for inconsistent formatting, spelling, and grammar:
	- Each link description starts with an uppercase character and ends with a period.
	- Drop all the `A` / `An` prefixes in the descriptions.
	- Consistent naming. For example, `Node.js`, not `NodeJS` or `node.js`.

In that spirit I created issues #29, #30 and #31. These should be completed before we close this issue.

Also, I don't think the image preposition is reasonable, I don't understand how it contributes to the project as a whole and also I have no design or imaging skills. It would just be a waste of time.

How does that sound to you?

**Comment by mlcdf** [comment]
You can assign me the #29 then. :smiley: 

And, I share your opinion regarding the logo/illustration.

**Comment by caramelomartins** [comment]
I see that you were able to assign it to yourself @mlcdf. Cool :+1:

**Comment by mlcdf** [comment]
Yep, thanks for adding me as a collaborator!

**Comment by caramelomartins** [comment]
Seems like this one is done now. Would you like to try and make the PR @mlcdf ?

**Comment by mlcdf** [comment]
Sure!

**Comment by caramelomartins** [comment]
Cool. I think you can close the issue when the PR is created. Either it gets accepted or it won't. :smile: 

Thank you for the effort! :+1:

**Comment by mlcdf** [comment]
Here the PR: https://github.com/sindresorhus/awesome/pull/858

Good work! :+1:

**Comment by mlcdf** [comment]
sindresorhus made a comment on the PR: 
> I would like to see more than just linters. Include some blog posts, tutorials, links to community resources (Stack Overflow tag, IRC, chat, forums, communities. Example: https://github.com/sindresorhus/awesome-electron#community), videos of talks, book, learning material, etc. Just a list of linters isn't that useful. Most language specific lists already document their linters.

His point is valid and I had not thought about it.

I'm pretty busy right now (it's the end of the semester), but I'll have some free time in February to do this. :)

**Comment by caramelomartins** [comment]
His point is valid but misguided. **I don't feel this is an issue and should not be treated as one.**

This list was not made with the goal of inserting it into sindresorhus/awesome. It was made to be useful for me as a repository for linting information and other people for the same reason.

As a suggestion this would be a good thing but as far as I'm concerned we should not be pushing to find random stuff to put here on this topic just so we can get into his project. This is a collection of useful information - we shouldn't be searching for it simply just to be part of that repository.

He made a suggestion and that is as valid as anything but he's coming from his own perspective, his own community and his feelings towards the use *he* could give to the list which isn't remotely connected to what the people that read this list feel. He's not looking at it from the perspective of the few people that look at this and think "this is a cool resource, I'm going to save it". I'm not even sure he could because of the massive size of his OS projects.

In this sense, if people find it useful to just have linters. Why wouldn't it just have linters?

I have no problem with the pull request being rejected on the grounds that he doesn't find the list useful but we won't try to push everything we can find in here just so we can make it on the list. I'll actually rather be rejected than do more of that like we did to get the list on a state that it could be made into a PR.

Now with this being said, I have no issue with expanding the project towards all those fields he mentioned but honestly, I won't do it to be part of his repository. We'll do it incrementally instead of just to be part of his repository.

**Comment by mlcdf** [comment]
Yeah you're right. I mean, I really like his (Sindre's) idea of expending the scope of this list and also including others resources in order to help people that aren't already familiar with linters. But, we should'nt do this just to get into the `awesome` list. So yeah, let's not rush this.

**Comment by caramelomartins** [comment]
Cool. Since we agree @mlcdf, I'm going to close both issues.

In the near feature (next week probably) I'll extend the scope of the project in the README.md.

Thanks!
